src/LibreCards.Tests/CardRepositoryTests.cs

```diff
@@ -0,0 +1,40 @@
+﻿using LibreCards.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace LibreCards.Tests
+{
+    public class CardRepositoryTests
+    {
+        private readonly ICardRepository _cardRepository;
+
+        public CardRepositoryTests()
+        {
+            _cardRepository = new CardRepository();
+        }
+
+        [Fact]
+        public void CardRepository_DrawOneCard()
+        {
+            var cards = _cardRepository.DrawCards();
+            Assert.Single(cards);
+        }
+        [Fact]
+        public void CardRepository_DrawManyCards()
+        {
+            var cards = _cardRepository.DrawCards(2).ToList();
+            cards.AddRange(_cardRepository.DrawCards(2));
+            Assert.Equal(4, cards.Count());
+            Assert.Equal(cards.Count(), cards.Distinct().Count());
+        }
+        [Fact]
+        public void CardRepository_DrawCardsOverLimit()
+        {
+            Assert.Throws<IndexOutOfRangeException>(() => _cardRepository.DrawCards(short.MaxValue));
```

Comment:
This is a test that does not work for me - it does not throw the error:
```
 Assert.Throws() Failure 
 Expected: typeof(System.IndexOutOfRangeException) 
 Actual: (No exception was thrown)
```
In any other test the exception would be thrown as expected (but not using `Assert.Throws` to execute the action).
`DrawCards` method does not execute method `GetResponseCards` called in the return statement and just skips it.

Reply:
Fixed in last commit, how? Magic (I have no idea, seriously).